### PR TITLE
Craco, yarn workspaces, and repo restructuring.

### PR DESCRIPTION
**Pull request by larskendall**
- Replaced `rescripts` with `craco`
- Switched from `npm` to `yarn` for package management
- Setup yarn workspaces
- Moved all client code into `client` folder
- Created `server/src` folder for consistency with client
- Updated shell scripts and Dockerfile for new project structure
- Removed some unnecessary/redundant shell scripts
- Updated project engine requirements for current NodeJS LTS version
- Cleaned up some formatting and eslint errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,22 +44,19 @@ USER_EDIPI=0000000001
...
-npm run dev
+yarn run seed-dev
+yarn run dev
 ```
 This will start both the frontend and the backend.
 
-In the logs you’ll see [0] and [1] at the start of each line.
-[0] is from the server and [1] is from the client.
-
 If you need to stop or reset the docker-compose dev database:
```
Comment (larskendall) [inline README.md:57]: In the new setup these now say `[server]` and `[client]`, and have pretty colors too!

File: client/src/components/query-builder/query-builder.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import {
   Select,
```
Comment (larskendall) [inline client/src/components/query-builder/query-builder.tsx:9]: All of the updates to this file were just eslint fixes. It should be safe to ignore.

File: client/src/index.tsx [hunk 3]
```diff
@@ -32,8 +31,3 @@ ReactDOM.render(
   </React.StrictMode>,
   document.getElementById('root'),
 );
-
-// If you want your app to work offline and load faster, you can change
-// unregister() to register() below. Note this comes with some pitfalls.
-// Learn more about service workers: https://bit.ly/CRA-PWA
-serviceWorker.unregister();
```
Comment (larskendall) [inline client/src/index.tsx:39]: The server worker was giving me errors in the new setup, and I don't think we were even using it, so it seemed worth removing.

File: client/src/store.ts [hunk 4]
```diff
@@ -123,7 +122,7 @@ const persistConfig = {
 
 export const configureStore = () => {
   const store = createStore(
-    persistReducer(
+    persistReducer<any>(
```
Comment (larskendall) [inline client/src/store.ts:125]: This was giving me compile errors and I couldn't figure out a proper type after quite a bit of effort, so just went with `any` for now.

File: package.json [hunk 5]
```diff
@@ -1,170 +1,39 @@
...
+    "concurrently": "^6.2.0",
+    "eslint": "^7.29.0",
+    "eslint-config-airbnb": "^18.2.1",
+    "eslint-config-airbnb-typescript": "^12.3.1",
+    "eslint-plugin-chai-friendly": "^0.7.1",
+    "eslint-plugin-import": "^2.23.4",
+    "eslint-plugin-jsx-a11y": "^6.4.1",
+    "eslint-plugin-mocha-no-only": "^1.1.1",
+    "eslint-plugin-react": "^7.24.0",
+    "eslint-plugin-react-hooks": "^4.2.0",
```
Comment (larskendall) [inline package.json:36]: The reason for all of the react eslint plugins at this top level is just that `eslint-config-airbnb-typescript` seems to have some hard dependencies on them (oddly). So adding them all here suppressed yarn install warnings.

File: server/src/api/observation/index.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import express from 'express';
```
Comment (larskendall) [inline server/src/api/observation/index.ts:1]: Just formatting/naming fixes in this file.

**Review by AnthonyWhitaker — APPROVED** [review]
Awesome stuff. I've been using this branch pretty much all day and it seems to be working fine. I ran into some issues on initial startups with needing to `chmod +x start-db.sh` and some other files - but since that's obviously not impacting the Windows users, I think it's okay as is.

**Comment by larskendall** [comment]
@AnthonyWhitaker Good catch! I'll chmod them and then merge.

Thanks for knocking this PR out so quickly!

**Pull request merged**